dndtroops-persistence/src/main/java/cz/muni/fi/pa165/dndtroops/entities/Troop.java

```diff
@@ -43,31 +38,26 @@ public Troop(){
      */
 
 
-    public Troop(long id, String name, String mission, long goldenMoney)
+    public Troop(String name, String mission, Long goldenMoney)
     {
-        this.id = id;
         this.name=name;
         this.mission=mission;
         this.goldenMoney=goldenMoney;
-
     }
 
     /*
     Override of default Object.equals() method
      */
 
-
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (!(o instanceof Troop)) return false;
 
         Troop troop = (Troop) o;
 
-        if (getId() != troop.getId()) return false;
-        if (getGoldenMoney() != troop.getGoldenMoney()) return false;
-        if (getName() != null ? !getName().equals(troop.getName()) : troop.getName() != null) return false;
-        return getMission() != null ? getMission().equals(troop.getMission()) : troop.getMission() == null;
+        if (!getName().equals(troop.getName())) return false;
+        return (getMission().equals(troop.getMission()));
```

Comment:
Does Troop's mission fully depends on the Troop's name? What are the minimal requirements for entity's business key?